src/pkgcheck/checks/codingstyle.py

```diff
@@ -1118,3 +1118,33 @@ def feed(self, pkg):
             lines.append(lineno)
         if lines:
             yield ExcessiveLineLength(lines=lines, pkg=pkg)
+
+
+class InstallCompressedManpage(results.LineResult, results.Warning):
+    """Compressed manpages are not supported by ``doman`` or ``newman``."""
+
+    def __init__(self, func, **kwargs):
+        super().__init__(**kwargs)
+        self.func = func
+
+    @property
+    def desc(self):
+        return f'line {self.lineno}: compressed manpage {self.line!r} passed to {self.func}'
+
+
+class InstallCompressedManpageCheck(Check):
+    """Scan ebuild for compressed manpages passed to ``doman`` or ``newman``."""
+
+    _source = sources.EbuildParseRepoSource
+    known_results = frozenset([InstallCompressedManpage])
+
+    def feed(self, pkg):
+        for node, _ in bash.cmd_query.captures(pkg.tree.root_node):
+            call_name = pkg.node_str(node.child_by_field_name('name'))
+            if call_name not in ('doman', 'newman'):
+                continue
+            for arg in node.children[1:]:
+                arg_name = pkg.node_str(arg).strip('\'\"')
+                lineno, _ = arg.start_point
+                if arg_name.endswith(('.gz', '.bz2', '.xz')):
```

Comment:
Technically the list could be much bigger, but it's quite unlikely upstreams would ship them with any other suffix.

The list at https://github.com/gentoo/portage/commit/636dd15437eccfaf63e76c36412999b5e118c209 is mostly complete fwiw.

Reply:
nice, thanks!